src/h.js

```diff
@@ -8,16 +8,18 @@ export function h (type, config) {
     if (vnode == null || vnode === true || vnode === false) {
     } else if (Array.isArray(vnode) || typeof vnode === 'object') {
       children.push(vnode)
-    } else if (typeof vnode === 'function') {
-      children = vnode
```

Comment:
Note: there is no special handling of `function` type literals in React - whether `children` is an array (or undefined) is determined by the number of JSX child nodes/literals, not by their types.